doc/user/metrics/README.md

```diff
@@ -86,3 +86,124 @@ By default operator will expose info metrics based on the number of the current
 [ownerref-permission]: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
 [ksm]: https://github.com/kubernetes/kube-state-metrics
 [controller-metrics]: https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/internal/controller/metrics
+
+
+### Custom Metrics using Operator-SDK
+To implement custom metric collection using the `metrics package` provided by SDK, follow the steps outlined below:
+1. Update the Operator: Update the operator to provide sufficient permisisons so that the required resources for prometheus can be created.
+2. Custom metric collection: Implement the logic for custom metric variables defined for your operator and register the specific metrics with prometheus. 
+4. Expose metrics: Utilize the functions provided by SDK to expose metrics and start the metrics server.
+
+#### Update the Operator
+
+Update the operator and provide sufficient permissions to allow creation of Prometheus resources. The required permissions are:
+1. `Service Monitor` : `get` and `create`
+2. `roles` and `rolebindings` : `get`, `create`, `delete`, `update`, `list`
+
+The below is a snippet from roles.yaml:
+```
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - get
+  - create
+  - delete
+  - update
+  - list
+  - watch
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
+  - get
+  - create
+
+```
+
+
+#### Custom metric collection
+
+Implement the logic to collect specific custom metrics inside the operator code base. For example, the following is a snippet from `Marketplace` operator which implements custom metric collection:
+
+```
+// Metric variables and metric list that are used to report error. 
+var (
+	errorMetric = prometheus.NewCounterVec(
```

Comment:
It shows very nice, but IMHO the indentation should equal it is done in the other docs. See here: https://github.com/operator-framework/operator-sdk/blob/master/doc/user/unit-testing.md

Reply:
Solved